src/main/java/io/rocketpartners/cloud/action/elastic/v03x/ElasticDb.java

```diff
@@ -109,9 +110,15 @@ private void reflectDb() throws Exception
       }
 
       // 'GET _all' returns all indices/aliases/mappings
-
-      FutureResponse future = HttpUtils.get(getUrl() + "/_all");
-      Response allResp = future.get(maxRequestDuration, TimeUnit.SECONDS);
+      String allRequestUrl = getUrl() + "/_all";
+      FutureResponse future = HttpUtils.get(allRequestUrl);
+      Response allResp;
+      try {
+          allResp = future.get(maxRequestDuration, TimeUnit.SECONDS);
+      } catch (TimeoutException e) {
+          log.error("timed out waiting {} seconds for GET {} to complete", maxRequestDuration, allRequestUrl);
+          throw e;
+      }
```

Comment:
might be a silly question but what is the relationship between these two classes? could this change be single-sourced by putting the common elements into a reusable method accessible to both classes?

Reply:
a very good question, I don't know the answer. At first glance, they do more/less the same thing.